assembl/tests/views_tests/test_routes.py

```diff
@@ -298,24 +299,26 @@ def test_route_discussion_idea_legacy(discussion, root_post_1, subidea_1,
 
 def test_route_discussion_post_v2(
         test_app, discussion_with_2_phase_interface_v2,
-        timeline_phase2_interface_v2,
-        post_related_to_sub_idea_1, subidea_1):
+        post_related_to_sub_idea_1, subidea_1, test_session):
 
     from assembl.lib.frontend_urls import FrontendUrls
+    from assembl import models
     slug = discussion_with_2_phase_interface_v2.slug
     route = "/%s/posts/%s" % (
         slug, quote_plus(post_related_to_sub_idea_1.uri()))
     print route
     resp = test_app.get(route)
     assert resp.status_int == 303
 
+    thread_phase = test_session.query(models.DiscussionPhase).all()[1]
```

Comment:
The order of such query is undefined, thread may not be in second position. You should add order field in your DiscussionPhase in your phases fixture and do an explicit `order_by(models.DiscussionPhase)` here or use your `phases` fixture with `phases['thread']`

Reply:
Just filter the query (DiscussionPhase.identifier == 'thread')